src/controllers/resource.ts

```diff
@@ -242,6 +275,146 @@ export class ResourceController {
 					);
 			}
 
+			options.network = options.network || (did.split(':')[2] as NetworkType);
+			await CheqdRegistrar.instance.connect(options);
+			const result = await CheqdRegistrar.instance.createResource(signInputs, resourcePayload);
+			if (result.code == 0) {
+				return response
+					.status(201)
+					.json(Responses.GetResourceSuccessResponse(jobId, secret, did, resourcePayload));
+			} else {
+				return response
+					.status(400)
+					.json(Responses.GetInvalidResourceResponse(did, resourcePayload, secret, Messages.InvalidResource));
+			}
+		} catch (error) {
+			return response.status(500).json({
+				jobId,
+				didUrlState: {
+					state: IState.Failed,
+					reason: Messages.Internal,
+					description: Messages.TryAgain + error,
+					secret,
+					resourcePayload,
+				},
+			});
+		}
+	}
+	public async updateResource(request: Request, response: Response) {
+		const result = validationResult(request);
+		if (!result.isEmpty()) {
+			return response
+				.status(400)
+				.json(Responses.GetInvalidResourceResponse('', {}, request.body.secret, result.array()[0].msg));
+		}
+
+		let {
+			did,
+			jobId,
+			content,
+			relativeDidUrl,
+			name,
+			type,
+			version,
+			secret = {},
+			options = {},
+		} = request.body as IResourceUpdateRequest;
+
+		let resourcePayload: Partial<MsgCreateResourcePayload> = {};
+		let existingResource;
+		try {
+			// check if did is registered on the ledger
+			let resolvedDocument = await CheqdResolver(did);
+			if (!resolvedDocument?.didDocument || resolvedDocument.didDocumentMetadata.deactivated) {
+				return response
+					.status(400)
+					.send(Responses.GetInvalidResourceResponse(did, {}, secret, Messages.DidNotFound));
+			} else {
+				const didUrlIndex = resolvedDocument.didDocumentMetadata.linkedResourceMetadata.findIndex(
+					(resource: { resourceURI: string }) => resource.resourceURI === did + relativeDidUrl
+				);
+				if (didUrlIndex !== -1) {
+					existingResource = resolvedDocument.didDocumentMetadata.linkedResourceMetadata[didUrlIndex];
+					if (existingResource.resourceName !== name || existingResource.resourceType !== type)
```

Comment:
also validate if nextVersionId is null. Because only the latest resource can be updated